src/wsteth-exchange-rate-adapter/WstEthStEthExchangeRateChainlinkAdapter.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.21;
+
+import {IStEth} from "./interfaces/IStEth.sol";
+import {MinimalAggregatorV3Interface} from "./interfaces/MinimalAggregatorV3Interface.sol";
+
+/// @title WstEthStEthExchangeRateChainlinkAdapter
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+/// @notice wstETH/stETH exchange rate price feed.
+/// @dev This contract should only be deployed on Ethereum and used as a price feed for Morpho oracles.
+contract WstEthStEthExchangeRateChainlinkAdapter is MinimalAggregatorV3Interface {
+    /// @inheritdoc MinimalAggregatorV3Interface
+    // @dev The calculated price has 18 decimals precision, whatever the value of `decimals`.
+    uint8 public constant decimals = 18;
+
+    /// @notice The description of the price feed.
+    string public constant description = "wstETH/stETH exchange rate";
+
+    /// @notice The address of stETH on Ethereum.
+    IStEth public constant ST_ETH = IStEth(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84);
+
+    /// @inheritdoc MinimalAggregatorV3Interface
+    /// @dev Returns zero for roundId, startedAt, updatedAt and answeredInRound.
+    /// @dev Silently overflows if `getPooledEthByShares`'s return value is greater than `type(int256).max`.
```

Comment:
If we want to be strictly consistent, we'd create an interface for this adapter and move the natspecs there

Reply:
So this would change the interface `MinimalAggregatorV3Interface` to a more consistent name, that is a name starting with `I` (note that it already is not exactly the name given by Chainlink)

Reply:
let's not bother with that

Reply:
>we'd create an interface for this adapter and move the natspecs there

the problem is that we are putting dev comments on the `MinimalAggregatorV3Interface` functions, so it would mean that this adapter no longer implements it ?

Reply:
>So this would change the interface MinimalAggregatorV3Interface to a more consistent name, that is a name starting with I (note that it already is not exactly the name given by Chainlink)

anyway we could have had a better name for it indeed

Reply:
I was more talking about another interface, specific to `WstEthStEthExchangeRateChainlinkAdapter`: `IWstEthStEthExchangeRateChainlinkAdapter`, implemented alongside `MinimalAggregatorV3Interface`

Reply:
seen with @Rubilmax, it would actually mean not implementing anymore MinimalAggregatorV3Interface. So we abandon the idea